packages/core/test/integration/hooks/incrementDecrement.test.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const chai = require('chai');
+
+const expect = chai.expect;
+const Support = require('../support');
+const { DataTypes } = require('@sequelize/core');
+
+describe(Support.getTestDialectTeaser('Hooks'), () => {
+  beforeEach(async function () {
+    this.User = this.sequelize.define('User', {
+      username: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      integer: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+      },
+    });
+    await this.sequelize.sync({ force: true });
+  });
+
+  describe('#incrementdecrement', () => {
+    beforeEach(async function () {
+      await this.User.bulkCreate([
+        { username: 'adam', integer: 1 },
+      ]);
+    });
+
+    describe('on success', () => {
+      it('before hook runs', async () => {
```

Comment:
there should be tests that verify which parameters and options can be modified by the hook